### PR TITLE
Use ESX transport for EL9 image when Host resource exists

### PR DESCRIPTION
**Pull request by nyoxi**
When there is a Host resource for the ESXi host on which the converted VM is located we use the `esx://` transport when running virt-v2v. The Host resource is normally created when migration network is configured in UI for the provider. There is not switch or option whether to use `vpx://` or `esx://` and the choice is based solely on the existence of the Host resource. This mimics the behavior that exists with CDI.

This also fixes a bug with EL9 image. When user configured migration network, e.g. to be used for warm migrations with CDI, this prevented any cold migrations because we passed wrong credentials in conversion pod secret and no thumbprint.

### CODE REVIEW

**Review by ahadas — APPROVED** [review]

File: pkg/controller/plan/adapter/vsphere/builder.go [hunk 1]
```diff
@@ -196,37 +201,52 @@ func (r *Builder) PodEnvironment(vmRef ref.Ref, sourceSecret *core.Secret) (env
...
-			path = parent.Path
+		libvirtURL = liburl.URL{
+			Scheme:   "esx",
+			Host:     hostDef.Spec.IpAddress,
+			User:     liburl.User(string(hostSecret.Data["user"])),
+			Path:     "",
+			RawQuery: sslVerify,
+		}
+	} else {
+		// Connecth through VCenter
```
Comment (ahadas) [inline pkg/controller/plan/adapter/vsphere/builder.go:221]:
```suggestion
		// Connect through VCenter
```

File: pkg/controller/plan/adapter/vsphere/builder.go [hunk 2]
```diff
@@ -247,19 +267,27 @@ func (r *Builder) Secret(vmRef ref.Ref, in, object *core.Secret) (err error) {
...
 			return
 		}
 		in = hostSecret
+		// Get host thumbprint
+		h, nErr := r.host(hostID)
+		if nErr != nil {
+			err = nErr
+			return
+		}
+		thumbprint = h.Thumbprint
```
Comment (ahadas) [inline pkg/controller/plan/adapter/vsphere/builder.go:284]: yeah, I think it's better the way you did it compared to set it on the secret here (in.Data["thumbprint"] = h.Thumbprint)

**Review by ahadas — COMMENTED** [review]
minor typo, other than that lgtm

File: pkg/controller/plan/adapter/vsphere/builder.go (hunk 2 above)
Comment (nyoxi) [reply pkg/controller/plan/adapter/vsphere/builder.go:284]: It's better not to change the source provider secret. It should always contain the VCenter thumbprint.

File: pkg/controller/plan/adapter/vsphere/builder.go (hunk 1 above)
Comment (nyoxi) [reply pkg/controller/plan/adapter/vsphere/builder.go]: Done

File: pkg/controller/plan/adapter/vsphere/builder.go (hunk 2 above)
Comment (ahadas) [reply pkg/controller/plan/adapter/vsphere/builder.go:284]: yeah, I meant without persisting the changes into etcd, just so we'll operate on the secret below - but I do like the way you wrote it more

**Review by ahadas — APPROVED** [review]

**Pull request merged**